connectivity/netsocket/source/NetworkInterface.cpp

```diff
@@ -142,22 +143,43 @@ static void call_all_event_listeners(NetworkInterface *iface, nsapi_event_t even
     }
 }
 
-void NetworkInterface::add_event_listener(mbed::Callback<void(nsapi_event_t, intptr_t)> status_cb)
+nsapi_error_t NetworkInterface::internal_add_event_listener(mbed::Callback<void(nsapi_event_t, intptr_t)> status_cb)
 {
     iface_eventlist_t *event_list = get_interface_event_list_head();
 #if MBED_CONF_PLATFORM_CALLBACK_COMPARABLE
     ns_list_foreach_safe(iface_eventlist_entry_t, entry, event_list) {
         if (entry->status_cb == status_cb && entry->iface == this) {
-            return;
+            return NSAPI_ERROR_OK;
         }
     }
 #endif
-    iface_eventlist_entry_t *entry = new iface_eventlist_entry_t;
+
+    iface_eventlist_entry_t *entry = new (std::nothrow) iface_eventlist_entry_t;
+    if (!entry) {
+        return NSAPI_ERROR_NO_MEMORY;
+    }
+
     entry->iface = this;
     entry->status_cb = status_cb;
     ns_list_add_to_end(event_list, entry);
     attach(mbed::callback(&call_all_event_listeners, this));
+    return NSAPI_ERROR_OK;
+}
+
+#if MBED_CONF_NSAPI_ADD_EVENT_LISTENER_RETURN_CHANGE
+nsapi_error_t NetworkInterface::add_event_listener(mbed::Callback<void(nsapi_event_t, intptr_t)> status_cb)
+{
+    return internal_add_event_listener(status_cb);
+}
+#else
+void NetworkInterface::add_event_listener(mbed::Callback<void(nsapi_event_t, intptr_t)> status_cb)
+{
+    auto error = internal_add_event_listener(status_cb);
+    if (error) {
+        MBED_ERROR(error, __func__);
```

Comment:
We're losing the scope with `__func__` so it just expand to `add_event_listener` which is pretty generic. Can you provide more information in the error message ?